### DISCUSSION TITLE
Plotting sparse matrices with markers?

### DISCUSSION THREAD

**Discussion by jhkliu42** [chanzuckerberg/napari-hub#455] [Q&A] [answered]
Hello napari folks,

I'm wondering if there's currently any feature in napari to plot scipy-style sparse matrices with markers, similar to the matplotlib.pyplot.spy() function (https://matplotlib.org/3.1.1/api/_as_gen/matplotlib.pyplot.spy.html).

I've been working with this for MERFISH data to quickly plot >1 million transcripts from xy coordinate positions by generating a sparse matrix with dimensions equal to the total number of pixels in the imaging experiment and setting the value of each pixel to 1 whenever there's a transcript detected within that pixel. Compared to matplotlib's scatterplot function or numpy's histogram2d function this works way faster, and I was wondering if napari currently has or could be modified to have a similar feature.

Thanks!
Jonathan

**Comment by neuromusic** [comment, marked as answer]
hi @jhkliu42! 

this is probably a better question for image.sc: https://forum.image.sc/tag/napari
